# See https://github.com/LudovicRousseau/PCSC/issues/187
diff -urNad pcsc-lite-2.2.1-orig/meson.build pcsc-lite-2.2.1/meson.build
--- pcsc-lite-2.2.1-orig/meson.build	2024-05-08 12:57:49.000000000 +0300
+++ pcsc-lite-2.2.1/meson.build	2024-05-21 19:14:52.950227430 +0300
@@ -100,6 +100,9 @@
 endif
 
 if get_option('libusb')
+  if get_option('libudev')
+    error('You can\'t use both libudev and libusb')
+  endif
   libusb_dep = dependency('libusb-1.0')
   pcscd_dep += libusb_dep
   conf_data.set('HAVE_LIBUSB', true)
@@ -145,6 +148,7 @@
   dependencies : pcscd_dep,
   c_args: '-DPCSCD',
   export_dynamic : true,
+  link_args: '-ldl',
   install_dir : sbindir,
   install : true)
 
@@ -175,6 +179,7 @@
   libpcsclite_src,
   include_directories : incdir,
   soversion : 1,
+  link_args: '-ldl',
   install : true)
 
 # libpcsclite_fake library
@@ -187,6 +192,7 @@
   sources : ['src/spy/libpcscspy.c', 'src/sys_unix.c'],
   include_directories : incdir,
   soversion : 0,
+  link_args: '-ldl',
   install : true)
 
 install_data('src/spy/pcsc-spy',
